[bitnami/etcd] Allow release namespace to be overridden

xgchena opened the pull request:
### Description of the change

This patch replaces `.Release.Namespace` by `common.names.namespace` that supports overriding namespace by `Values.namespaceOverride`.

### Benefits

It enables deploying the chart in a custom namespace.

### Possible drawbacks

N/A. (The fix is backward-compatible because `common.names.namespace` defaults to `.Release.Namespace` if `Values.namespaceOverride` is not provided.)

### Applicable issues

N/A. (Searched for namespaceOverride, etcd and no related issues found.)

### Additional information

> Please run any tests or examples that can exercise your modified code

Installed the new chart on a k8s cluster with or without namespaceOverride, verified that both works, 

* With namespaceOverride

```console
$ helm dependency build bitnami/etcd

$ kubectl create namespace etcd-system

$ helm upgrade --install etcd-release bitnami/etcd --values test-values.yaml --set namespaceOverride=etcd-system

$ kubectl get all -A --all-namespaces |grep etcd
etcd-system    pod/etcd-release-0    1/1    Running    0    22s
etcd-system    service/etcd-release     ClusterIP   10.100.225.33    <none>    2379/TCP,2380/TCP     23s
etcd-system    service/etcd-release-headless    ClusterIP   None    <none>    2379/TCP,2380/TCP    23s
etcd-system    statefulset.apps/etcd-release    1/1    23s

$ kubectl run etcd-client -it --rm --image docker.io/bitnami/etcd:3.5.21-debian-12-r1 --env ETCDCTL_ENDPOINTS="etcd-release.etcd-system.svc.cluster.local:2379" --command -- bash
I have no name!@etcd-client:/opt/bitnami/etcd$ etcdctl put /message Hello
OK
I have no name!@etcd-client:/opt/bitnami/etcd$ etcdctl get /message
/message
Hello
I have no name!@etcd-client:/opt/bitnami/etcd$ exit

$ helm uninstall etcd-release
$ kubectl delete namespace etcd-system
```

* Without namespaceOverride

```console
$ helm upgrade --install etcd-release bitnami/etcd --values test-values.yaml

$ kubectl get all -A --all-namespaces |grep etcd
default    pod/etcd-release-0    1/1    Running   0    10s
default    service/etcd-release    ClusterIP   10.100.126.230   <none>    2379/TCP,2380/TCP    10s
default    service/etcd-release-headless    ClusterIP   None    <none>    2379/TCP,2380/TCP    10s
default    statefulset.apps/etcd-release   1/1    10s

$ kubectl run etcd-client -it --rm --image docker.io/bitnami/etcd:3.5.21-debian-12-r1 --env ETCDCTL_ENDPOINTS="etcd-release.default.svc.cluster.local:2379" --command -- bash
If you don't see a command prompt, try pressing enter.
I have no name!@etcd-client:/opt/bitnami/etcd$ etcdctl put /message Hello
OK
I have no name!@etcd-client:/opt/bitnami/etcd$ etcdctl get /message
/message
Hello
I have no name!@etcd-client:/opt/bitnami/etcd$ exit
exit

$ helm uninstall etcd-release
```

### Checklist

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [X] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami/readme-generator-for-helm)
- [X] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

gongomgra commented:
Hi @xgchena 

Thanks for your contribution and sorry for the delay on getting back to you. Can you update your branch and fix the conflicts so we can merge your changes?

xgchena commented:
> Hi @xgchena
> 
> Thanks for your contribution and sorry for the delay on getting back to you. Can you update your branch and fix the conflicts so we can merge your changes?

Thanks for reviewing the changes. I have updated the branch:
* rebased from the latest main branch and resolved conflicts. 
* confirmed that there is no occurrence of `.Release.Namespace` (by `grep -re "\.Release\.Namespace" bitnami/etcd`)
* re-executed the test steps in the PR description
* bumped up bitnami/etcd chart version to 11.3.6
